packs/autoscale/sensors/autoscale_governor_sensor.py

```diff
@@ -75,12 +81,18 @@ def _process_asg(self, asg, action):
         trigger_type         = self._trigger[action]
         bound                = self._bound[action]
 
+        group_status         = self._kvp_get('asg.%s.status' % (asg), local=False)
         last_event_timestamp = self._kvp_get('asg.%s.last_%s_timestamp' % (asg, action), local=False)
         event_delay          = self._kvp_get('asg.%s.%s_delay' % (asg, action), local=False)
         current_node_count   = self._kvp_get('asg.%s.total_nodes' % (asg), local=False)
         node_bound           = self._kvp_get('asg.%s.%s_nodes' % (asg, bound), local=False)
         total_nodes          = self._kvp_get('asg.%s.total_nodes' % (asg), local=False)
 
+        if group_status in GROUP_ACTIVE_STATUS:
```

Comment:
I check for "active" statuses and not for idle status here to avoid potential dead-locking.

When an expansion / deflation process is triggered, I set the status to expanding / deflating and set the TTL on this item. This means that if this process for some reason crashes or doesn't complete in TTL seconds, this item will be removed and governor will continue with emitting the up / down triggers.

If I wouldn't set the TTL, expansion / deflation work-flow crashing or not finishing would means group would stay in an "active" status which would result in a dead-lock (governor would never emit new events).